### DISCUSSION TITLE
Other than addRow functionality in addRowsComponent ?

### DISCUSSION THREAD

**Discussion by ronniee-07** [nick-keller/react-datasheet-grid#221] [Q&A] [answered]
Hi Nick,
I tried to add different button in addRowsComponent. But on click addRow() method is called. Is there anyway I can implement Undo, Update button just after addRow button ?

<img width="408" alt="Screenshot 2022-12-09 at 7 47 26 PM" src="https://user-images.githubusercontent.com/55025907/206722069-163a074c-238b-4796-a1c6-9b6efe54b806.png">

**Comment by nick-keller** [comment]
Can you share the code of your component ?
I'd assume you added a prop `onClick={addRow}` on the other buttons.

**Reply by ronniee-07** [reply]
<img width="823" alt="Screenshot 2022-12-09 at 9 26 54 PM" src="https://user-images.githubusercontent.com/55025907/206741770-e283b776-1b72-4522-9c1f-edae6dfc9e58.png">

**Reply by ronniee-07** [reply]
whenever I clicked on undo/update button already added row becomes empty & only one row with initial state will be there.

**Reply by nick-keller** [reply]
I tried it myself, this must come from the implementation of the `handleUpdate` and `handleUndo`. I suspect they might be wrapped in a useCallback or something, and you are missing a dependency on the value.

Feel free to share your implementation.

**Reply by ronniee-07** [reply]
Hi Nick, 
Please find below sandbox link. Behaviour is reproducible. Please correct me if i am doing anything wrong.
Thanks

https://codesandbox.io/s/ecstatic-gould-y0gv3c

**Comment by nick-keller** [comment, marked as answer]
Alright, as I suspected, your buttons don't have access to the "latest" versions of commit and cancel. 
What you have to do is create a ref:
<img width="724" alt="CleanShot 2022-12-20 at 14 00 08@2x" src="https://user-images.githubusercontent.com/2682443/208673566-9cd4b641-94fd-4925-9828-9cf3f15ef3fd.png">

**Reply by ronniee-07** [reply]
Thanks
